src/jit/lowerxarch.cpp

```diff
@@ -1989,6 +2024,14 @@ Lowering::TreeNodeInfoInitLogicalOp(GenTree* tree)
         assert(operand != nullptr);
         MakeSrcContained(tree, operand);
     }
+    else if (!binOpInRMW)
+    {
+        // If this binary op is neither have contained operands nor part of a
+        // Read-modify-write (RMW) operation, it is safe to mark its operands
+        // as reg optional.  This is to ensure that both the operands don't
+        // end up in memory.
+        SetRegOptionalForBinOp(tree);
```

Comment:
This comment needs rewording. First, the initial sentence should probably be: "If this binary op neither has contained operands, nor is a read-modify-write (RMW) operation" (I don't think it is correct to say the tree is PART of an RMW operand - either it is RMW or it is not, but perhaps you were referring to the operands).
Second, when you say "This is to ensure", it seems like you are referring to the marking of the operands as reg optional - but I think you mean that NOT marking them it is what ensures that they don't both end up in memory.